### PR TITLE
AMP Slidescroll : Replace inline-block with flex + Work around for scroll snap webkit bug

### PR DESCRIPTION
**Pull request by sriramkrish85**
**In this PR**
- AMP Slidescroll : Replace inline-block with flex
- Workaround for https://bugs.webkit.org/show_bug.cgi?id=158821

**Yet to come**

- Support for loops 
- Scroll Support for snap-points
- Polyfill snapping

### CODE REVIEW

File: extensions/amp-carousel/0.1/amp-carousel.css [hunk 1]
```diff
@@ -78,10 +78,13 @@ amp-carousel .amp-carousel-button.amp-disabled {
 }
 
 .-amp-slides-container {
+  align-items: center;
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: Why `align-items` and `text-align` set to `center`?

File: extensions/amp-carousel/0.1/amp-carousel.css [hunk 2]
```diff
@@ -90,14 +93,35 @@ amp-carousel .amp-carousel-button.amp-disabled {
 
 .-amp-slide-item {
   display: none;
+  flex: 1 0 100% !important;
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: Shouldn't it be `0 0 100%`? We don't want it to shrink or grow from basis, right?

File: extensions/amp-carousel/0.1/amp-carousel.css [hunk 3]
```diff
@@ -90,14 +93,35 @@ amp-carousel .amp-carousel-button.amp-disabled {
...
   height: 100% !important;
   position: relative;
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: `position: relative` must also be `!important`

File: extensions/amp-carousel/0.1/amp-carousel.css [hunk 4]
```diff
@@ -90,14 +93,35 @@ amp-carousel .amp-carousel-button.amp-disabled {
...
-  scroll-snap-coordinate: 0 0;
+  scroll-snap-coordinate: 0 0 !important;
   width: 100% !important;
 }
 
 .-amp-slide-item-show {
-  display: inline-block !important;
+  display: block !important;
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: Didn't we discuss that we'll try to make items flex as well? Or are we planning on trying it out later?

File: extensions/amp-carousel/0.1/amp-carousel.css [hunk 5]
```diff
@@ -90,14 +93,35 @@ amp-carousel .amp-carousel-button.amp-disabled {
...
+}
+
+.-amp-carousel-start-marker,
+.-amp-carousel-end-marker {
+  display: block !important;
+  height: 100% !important;
+  width: 1px !important;
+  background-color: transparent !important;
+  scroll-snap-coordinate: 0 0 !important;
+  flex: 1 0 1px !important;
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: also `0 0 1px`

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 6]
```diff
@@ -42,6 +42,13 @@ export class AmpSlideScroll extends BaseCarousel {
     this.slidesContainer_ = this.win_.document.createElement('div');
     this.slidesContainer_.classList.add('-amp-slides-container');
 
+    // workaround - https://bugs.webkit.org/show_bug.cgi?id=158821
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/slidescroll.js]: Capitalize

File: extensions/amp-carousel/0.1/amp-carousel.css [hunk 7]
```diff
@@ -90,14 +93,35 @@ amp-carousel .amp-carousel-button.amp-disabled {
...
+}
+
+.-amp-carousel-start-marker {
+  order: -1 !important;
+  margin-left: -1px !important;
+}
+
+.-amp-carousel-end-marker {
+  order: 100000000 !important;
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: 👍

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 8]
```diff
@@ -53,6 +60,14 @@ export class AmpSlideScroll extends BaseCarousel {
       this.slidesContainer_.appendChild(slideWrapper);
       this.slideWrappers_.push(slideWrapper);
     });
+
+    // workaround - https://bugs.webkit.org/show_bug.cgi?id=158821
+    if (this.hasNativeSnapPoints_) {
```
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/slidescroll.js]: With `order`, this `if` can be combined with the `-start-marker` above, right?

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 1 above)
Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/amp-carousel.css]: yes

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 4 above)
Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/amp-carousel.css]: I don't see any benefits as such. But yes we can try it out later. This should not affect the behavior.

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 5 above)
Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/amp-carousel.css]: Done

File: extensions/amp-carousel/0.1/slidescroll.js (hunk 6 above)
Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/slidescroll.js]: Done

File: extensions/amp-carousel/0.1/slidescroll.js (hunk 8 above)
Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/slidescroll.js]: yes - good catch.

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 3 above)
Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/amp-carousel.css]: Done

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 2 above)
Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/amp-carousel.css]: correct

**Comment by sriramkrish85** [comment]
PTAL

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: The question was "why" :)

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 2 above)
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: Which one "correct"?

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 4 above)
Comment (dvoytenko) [inline extensions/amp-carousel/0.1/amp-carousel.css]: Ok. Let's try it out later. The benefit would be that the slide's contents are centered.

**Comment by dvoytenko** [comment]
@sriramkrish85 all looks good. I only don't understand why the container does centering of any kind.

File: extensions/amp-carousel/0.1/amp-carousel.css (hunk 1 above)
Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/amp-carousel.css]: oh :) it also centers the content  (without needing to make it flex.)

Comment (sriramkrish85) [inline extensions/amp-carousel/0.1/amp-carousel.css]: actually i was wrong . Converting the slides to flex too.

**Comment by sriramkrish85** [comment]
PTAL

**Comment by dvoytenko** [comment]
LGTM

**Pull request merged**